packages/website/docs/sources.md

```diff
@@ -236,25 +260,158 @@ const autocomplete = createAutocomplete({
 });
 ```
 
-This pattern can be used to store data in the [context](context) before returning the sources.
+:::note
+
+You can use the official [`autocomplete-plugin-query-suggestions`](createQuerySuggestionsPlugin) plugin to retrieve Query Suggestions from Algolia.
+
+:::
+
+### Customizing items with templates
+
+In addition to defining data sources for items, a source also lets you customize how to display items using [`templates`](#templates-specific-to-algoliaautocomplete-js).
+
+:::info
+
+Templates are only supported in [autocomplete-js](autocomplete-js).
+
+:::
+
+Templates can return a string:
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return `Result: ${item.name}`;
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+Or a [Preact](https://preactjs.com/) component:
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+import { h } from 'preact';
+
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return h('div', null, item.name);
+          },
+        },
+      },
+    ];
+  },
+});
+
+```
+
+Or HTML/JSX-like syntax (here using [htm](https://github.com/developit/htm)):
+
+```js
+import { autocomplete, highlightHit } from '@algolia/autocomplete-js';
+import { html } from 'htm/preact';
+
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return html`<div>
+              ${html([highlightHit({ hit: item, attribute: 'name' })])}
+            </div>`
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+Or a custom component using `createElement` and `Fragment`:
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item, createElement, Fragment }) {
+            return createElement(Fragment, {}, item.name);
+          },
```

Comment:
In [this sandbox](https://codesandbox.io/s/algoliajs-example-forked-298f6) I've found that `createElement` and `Fragment` are indeed exposed by `item`, however it doesn't seem that `root` is exposed.

According to the shape of [`AutocompleteSource`](https://github.com/algolia/autocomplete.js/blob/next/packages/autocomplete-js/src/types/AutocompleteSource.ts#L20-L24) there should be access to `root`, however I'm unsure where `createElement` and `Fragment` are coming from.

Reply:
They're coming from #381 (not yet merged). We can quickly sync to make it clearer.